oodeel/extractor/torch_feature_extractor.py

```diff
@@ -317,6 +335,50 @@ def hook(_, __, output):
 
         return hook
 
+    def _get_scale_hook(self, percentile: float) -> Callable:
+        """
+        Hook that truncate activation features under a threshold value
+
+        Args:
+            threshold (float): threshold value
+
+        Returns:
+            Callable: hook function
+        """
+
+        def hook(_, __, output):
+            output_percentile = torch.quantile(output, percentile, dim=1)
+            mask = output > output_percentile[:, None]
+            output_masked = output * mask
+            s = torch.exp(torch.sum(output, dim=1) / torch.sum(output_masked, dim=1))
+            s = torch.unsqueeze(s, 1)
```

Comment:
Same suggestion, avoid duplicate code between both hooks?

Reply:
I agreed for keras because the duplicate code was longer and within the same method, but in that case, it is not so long, and I think it would hurt readability

Reply:
Yes, I agree for torch. Let's keep it that way